WordPress/gutenberg
# [Widgets Editor] Ensure all widgets are properly initialized when they're added, do not unmount widgets once they're mounted

## Description
At the moment, collapsing a widget area in the editor removes all DOM nodes inside. This breaks features like TinyMCE. What's more, newly added widgets are not propely initialized.

This PR is a first attempt at fixing that by
1. Preserving the DOM of collapsed widget areas
2. Rendering required form fields for newly added widgets

Solves #25551

**Notes:**
Updating `Panel` component could be more appropriate than the solution proposed in this PR. It also sounds like a larger endeavor - we'd likely need a new `<Hidden>` component, talk about the naming (we already have `<VisuallyHidden>`) and the most appropriate a11y way of hiding things. It's something we should do, but maybe as a follow up to this PR, it's not a blocking problem.

## How has this been tested?
1. Comment `'WP_Widget_Text',` in widgets.php
1. Go to the new widgets editor
1. Add a text widget, confirm TinyMCE was initialized
1. Collapse/uncollapse that widget area
1. Confirm TinyMCE was preserved
1. Click save, confirm it worked as it should

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
